### PR TITLE
Add Mpi suttport to 1-tree lower bound

### PR DESCRIPTION
**Pull request by johann-cm**
added mpi and rayon parallelism to the 1-tree lower bound

### CODE REVIEW

**Comment by johann-cm** [comment]
I've manually verified, that the MPI variant produces the same results as the rayon or sequential variant of the 1-tree lower bound

File: src/one_tree.rs [hunk 1]
```diff
@@ -88,6 +97,65 @@ fn one_tree_lower_bound_seq(graph: &NAMatrix) -> f64 {
...
+                SystemOperation::min(),
+            );
+        } else {
+            root_process.reduce_into(&local_lower_bound, SystemOperation::min());
+        }
+        global_lower_bound
+    }
+
+    // do not drop the Universe until the lower bound function has finished!
+    if let Some(universe) = mpi::initialize() {
```
Comment (lquenti) [inline src/one_tree.rs:150]: how does this work?

**Review by lquenti — APPROVED** [review]
lgtm

File: src/one_tree.rs (hunk 1 above)
Comment (lquenti) [reply src/one_tree.rs:150]: ah, understood now

**Pull request merged**